### PR TITLE
Task 77 -- BE: Refactored clockin and clock out

### PR DESCRIPTION
**Pull request by ChanM0**
# Description
refactored clockin and clock out
added new migrations to fit the new db design
Please include a summary of the change and which issue is fixed. 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/Http/Controllers/Api/TimeLog/ClockOutDomain/Services/ClockOutService.php [hunk 1]
```diff
@@ -1,66 +1,33 @@
...
-        $clockOutUUid = new UUID($this->domainName);
-        
-        $timeStamp = new TimeStamp(new UUID('timeStamp'), $timeStamp);
-
-        $clockOut = new ClockOut($clockOutUUid, $timeStamp);
-
-        return $clockOut;
+    public function __construct(ClockOutLogicContract $clockOutLogicUtility)
+    {
+        $this->clockOutLogicUtility = $clockOutLogicUtility;
```
Comment (JayJayAquino) [inline app/Http/Controllers/Api/TimeLog/ClockOutDomain/Services/ClockOutService.php:19]: variable names are different here

File: app/Http/Controllers/Api/TimeLog/ClockInDomain/Services/ClockInService.php [hunk 2]
```diff
@@ -1,96 +1,32 @@
...
-        $logParam['timeSheetId'] = $this->getTimeSheetId($user);
-        
-        $logParam['uuid'] = new UUID($this->domainName);
-
-        $timeStamp = new TimeStamp(new UUID('timeStamp'), $timeStamp);
-        
-        $logParam['clockIn'] = new ClockIn(new UUID('clockIn'), $timeStamp);
-        
-        return $logParam;
+        $this->clockInLogic = $clockInLogic;
```
Comment (JayJayAquino) [inline app/Http/Controllers/Api/TimeLog/ClockInDomain/Services/ClockInService.php]: variables referring to contracts should be consistent throughout the app. We probably need to decide on a universal term for this

File: app/Http/Controllers/Api/TimeLog/ClockInDomain/Services/ClockInService.php [hunk 3]
```diff
@@ -1,96 +1,32 @@
...
-use App\TimeSheets;
 
-//Exception
-use App\Exceptions\TimePuncherExceptions\ClockIn\AlreadyClockedIn;
 // Contracts
 use App\Http\Controllers\Api\TimeLog\ClockInDomain\Contracts\ClockInContract;
-use App\Http\Controllers\Api\TimeLog\TimePuncher\Contracts\TimePuncherContract;
+use App\Http\Controllers\Api\TimeLog\Logic\Contracts\ClockInLogicContract;
 
 class ClockInService implements ClockInContract
```
Comment (JayJayAquino) [inline app/Http/Controllers/Api/TimeLog/ClockInDomain/Services/ClockInService.php:11]: This service could possibly contain all the logic for clockin as opposed to 2 different services. Same goes for clockout.

Also the logic here could potentially be within the controller. Same goes for clockout

**Review by JayJayAquino — CHANGES_REQUESTED** [review]
See comments

**Review by chippingcolt — APPROVED** [review]
the green looks dope

**Review by JayJayAquino — APPROVED** [review]

**Pull request merged**